linalg/sparsemat.cpp

```diff
@@ -28,50 +28,58 @@ namespace mfem
 
 using namespace std;
 
-#ifdef MFEM_USE_CUDA
+#ifdef CUDA_or_HIP
```

Comment:
This macro name should be changed, maybe to a new macro `MFEM_USE_GPU`?

Reply:
I worry about `MFEM_USE_GPU` since CUDA and HIP are so close but other models might not match as well, esp for these sparse mat calls. @artv3 suggested `MFEM_USE_CUDA_OR_HIP`, which would be safe to use here, what do you think?